lib/engine.py

```diff
@@ -564,12 +566,15 @@ def format_insert_value(self, value, datatype, escape=True):
             else:
                 return "null"
         elif datatype in ("double", "decimal"):
-            if strvalue:
+
+            if strvalue and strvalue.strip() != "":
```

Comment:
Just to be clear (and I'm pretty sure you know this already, but wanted to avoid any confusion) empty strings return `False` when evaluated as a condition.